Source/SessionManager.swift

```diff
@@ -328,25 +362,18 @@ open class SessionManager {
 
     // MARK: Private - Download Implementation
 
-    private func download(
-        _ downloadable: Downloadable,
-        to destination: Request.DownloadFileDestination)
-        -> Request
-    {
+    private func download(_ downloadable: Downloadable, to destination: Request.DownloadFileDestination) -> Request {
         var downloadTask: URLSessionDownloadTask!
 
         switch downloadable {
-        case .request(let request):
-            queue.sync {
-                downloadTask = self.session.downloadTask(with: request)
-            }
-        case .resumeData(let resumeData):
-            queue.sync {
-                downloadTask = self.session.downloadTask(withResumeData: resumeData)
-            }
+        case let .request(urlRequest):
+            let urlRequest = adapt(urlRequest)
```

Comment:
Might be a bit clearer here if it was `let adaptedRequest = adapt(urlRequest)` so there's so confusion between the incoming parameter and what's actually requested. Plus it matches some other areas where the adapted request is named `adaptedRequest`.